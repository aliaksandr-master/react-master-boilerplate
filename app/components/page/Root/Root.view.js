import React from 'react';
import PropTypes from 'prop-types';
import { Slot, Router } from 'react-composite-router';
import LoadingBar from '../../ui/LoadingBar';
import PageNotFound from '../PageNotFound';
import './Root.view.less';



const Root = ({ children, history, routes, onStateChange }) => (
  <Router history={history} onChange={onStateChange} routes={routes}>
    <div className="Root">
      <LoadingBar selfID="global" />
      <Slot name="root">
        <div>{children ? children : <PageNotFound />}</div>
      </Slot>
    </div>
  </Router>
);



Root.propTypes = {
  children: PropTypes.node,
  routes: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  onStateChange: PropTypes.func // from container
};



Root.defaultProps = {
};



export default Root;
