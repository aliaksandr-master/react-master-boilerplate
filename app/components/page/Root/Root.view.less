@import '../../../lib/mixins.less';

.Root {
  // ._clearfix();
  height: 100%;
  min-height: 100%;
}
