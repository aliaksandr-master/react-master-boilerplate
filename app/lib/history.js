import createBrowserHistory from 'history/createBrowserHistory';
import { BASE_URL } from '../config';



export default createBrowserHistory({
  basename: BASE_URL
});
