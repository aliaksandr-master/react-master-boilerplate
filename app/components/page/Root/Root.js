import { compose, withHandlers } from 'recompose';
import update, { callWhenAreInitialized, callWhenPropsAreUpdated } from 'react-updating-hoc';
import { changeStateAction } from 'react-composite-router/lib/redux/reducer';
import { connect } from 'react-redux-self';
import { disableMainPreloader } from '../../../lib/preloader';
import App from './Root.view';



export default compose(
  connect({
    getters: [
      (_1, { loading: { loadingOnce } }) => loadingOnce
    ],
    selector: (loadingOnce) => ({
      loadingOnce
    }),
    mapDispatchToProps: {
      changeStateAction
    }
  }),
  withHandlers({
    onStateChange: ({ changeStateAction }) => (name, params) => {
      changeStateAction(name, params);
    }
  }),
  update(
    callWhenAreInitialized(() => {
      disableMainPreloader();
    }),
    callWhenPropsAreUpdated([ 'loadingOnce' ], () => {
      disableMainPreloader();
    })
  )
)(App);
