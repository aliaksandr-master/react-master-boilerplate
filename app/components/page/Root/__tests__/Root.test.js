/* eslint-env browser, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Root from '../index';
import Test from '../../../test/Test';
import routesTree from '../../../../routes/index';
import history from '../../../../lib/history';



describe('rendering', () => {
  it('should render something', () => {
    const component = renderer.create(
      <Test>
        <Root routes={routesTree.getRoutes()} history={history} />
      </Test>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
