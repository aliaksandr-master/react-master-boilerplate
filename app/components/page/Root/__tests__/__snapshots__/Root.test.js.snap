// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render something 1`] = `
<div
  className="Root"
>
  <div>
    <pre>
      {"style":{"position":"fixed","display":"none","top":0,"left":0,"right":0,"height":10,"zIndex":9999,"boxShadow":"0 0 3px rgba(0,0,0,0.25)"},"mode":"indeterminate"}
    </pre>
    <div />
  </div>
</div>
`;
